src/pydicom/pixels/decoders/base.py

```diff
@@ -1098,6 +1130,10 @@ def _as_array_encapsulated(runner: DecodeRunner, index: int | None) -> "np.ndarr
             original_nr_frames = runner.number_of_frames
             for frame in frame_generator:
                 if len(frame) == runner.frame_length(unit="bytes"):
+                    if original_bits_allocated == 1 and runner.get_option(
+                        "is_bitpacked"
+                    ):
+                        frame = unpack_bits(frame)[:pixels_per_frame]
```

Comment:
Doesn't this return an `ndarray` by default, but `frame` should be `bytes | bytearray` for `np.frombuffer()`. I probably should've called the `frame` variable `buffer` instead to make it clearer.

```python
buffer: bytes | bytearray
for buffer in frame_generator:
    if len(buffer) == runner.frame_length(unit="bytes"):
         if original_bits_allocated == 1 and runner.get_option("is_bitpacked"):
            frame = unpack_bits(buffer, as_array=True)[:pixels_per_Frame]
         else:
            frame = np.frombuffer(buffer, runner.pixel_dtype))
```

Reply:
This also works because of the behaviour of `np.frombuffer` when an array is passed. Refactored in [6d97c96](https://github.com/pydicom/pydicom/pull/2216/commits/6d97c96a2bb7d5b77dfe3ee9baacc97bd5f08699)  to be more explicit.